controller/gen/apis/server/v1beta1/types.go

```diff
@@ -29,7 +29,7 @@ type Server struct {
 
 // ServerSpec specifies a Server resource.
 type ServerSpec struct {
-	PodSelector   *metav1.LabelSelector `json:"podSelector,omitempty"`
```

Comment:
If this is generated code, I'm not sure that it's a great idea to modify it... Is there any way to annotate the CRD so that this will be set when regenerated?

What do the `Service` or `Deployment` resources do wrt label selectors and `omitempty`?

Reply:
@kleimkuhler do you know how this file was created? It's not created by `bin/update-codegen.sh` apparently.
The selectors for `Deployment` and `NetworkPolicy` aren't tagged with `omitempty`, but the one for `Service` is.

Reply:
This is not a generated file.  Rather, it is read by the code-gen scripts to generate other files.  (notice the lack of `// Code generated by client-gen. DO NOT EDIT.` comment)

Reply:
Still, are we sure we want to make this change? This will always cause both of the matches to be encoded, which isn't really necessary since we're changing the CRD...

Reply:
With `omitEmpty`, an empty `podSelector` is removed from the yaml generated by this type and the policy validator will reject it. Removing `omitEmpty` allows us to have `podSelector: {}`, with  `MatchLabels` and `MatchExpressions` not being encoded because they on the other hand still are tagged with `omitempty`.

Reply:
Ah, yeah. Makes sense!